### PR TITLE
NTI-9325: Prevent stripping of html data related to mentions and tagging.

### PR DESCRIPTION
**Pull request by hagenrd**

### CODE REVIEW

**Review by jamadden — CHANGES_REQUESTED** [review]
Hardcoding these here seems wrong. Can we find a more generic, extensible solution? Maybe query for an optional utility, adapter, or subscription adapter to add additional allowed attributes?

This will also need a change note.

File: src/nti/contentfragments/html.py [hunk 1]
```diff
@@ -144,6 +146,12 @@ def __init__(self, *args, **kwargs):
             'style',
             'xml:lang'
         ])
+
+        allowed_attr_provider = component.queryUtility(IAllowedAttributeProvider)
+        if allowed_attr_provider:
+            additional_attrs_allowed = frozenset(allowed_attr_provider.allowed_attributes)
+            allowed_attributes = allowed_attributes | additional_attrs_allowed
```
Comment (cutz) [inline src/nti/contentfragments/html.py:153]: If the attributes from the utility are going to be additive, we should probably make that clear in the interface docs.

File: src/nti/contentfragments/interfaces.py [hunk 2]
```diff
@@ -571,3 +576,16 @@ class ITagField(IPlainTextLineField):
...
+class IAllowedAttributeProvider(interface.Interface):
+    """
+    A way to provide a whitelist of attribute names that would be allowed
+    while parsing a content fragment.
+
+    .. versionadded:: 1.3.2
+    """
+
+    allowed_attributes = IndexedIterable(title=u"An iterable of attribute names allowed in a particular context",
+                                         value_type=TextLine(title=u"The attribute name"),
```
Comment (jamadden) [inline src/nti/contentfragments/interfaces.py]: I'm not sure those are TextLines. They look like NativeStringLines to me.

This should probably become version 1.4.

**Review by jamadden — APPROVED** [review]
A minor comment about types and versions from me, plus cutz's comment, and it looks good.

**Comment by jamadden** [comment]
I've pushed 1.4.0 to PyPI.

**Comment by hagenrd** [comment]
Much appreciated.

**Pull request merged**